src/components/container/ConnectDevicesMenu/ConnectDevicesMenu.tsx

```diff
@@ -422,4 +445,35 @@ function AppleHealthMenuItem(props: AppleHealthMenuItemProps) {
             </div>
         }
     </div>;
+}
+
+function WeatherMenuItem(props: { settings: DataCollectionSettings | null, participantInfo: ParticipantInfo | null, postalCodeSurveyName: string }){
+    let action = () => {
+        MyDataHelps.startSurvey(props.postalCodeSurveyName);
+    }
+
+    const postalCodeCustomFields = Object.keys(props.participantInfo ? props.participantInfo.customFields : {}).filter( (k) => k.endsWith('PostalCode'));
+    const postalCodes = [...postalCodeCustomFields.map((cf) => props.participantInfo?.customFields[cf]), props.participantInfo?.demographics.postalCode];
+    const postalCode = postalCodes.find((p) => !!p);
+
+    const indicator = postalCode ? 
+        <div className="mdhui-connect-devices-menu-connect">{postalCode}</div> :
+        <div className="mdhui-connect-devices-menu-connect">{language("setup")}</div>;
+
+    const titleBits = [];
+    if(props.settings?.airQualityEnabled) {
+        titleBits.push('Air Quality');
+    }
+    if(props.settings?.weatherEnabled) {
+        titleBits.push('Weather');
+    }
+    const title = titleBits.join(" / ");
+
+    return (
+        <div className="mdhui-connect-devices-menu-device">
+            <Action onClick={action} indicator={indicator}>
+                <Title autosizeImage image={<FontAwesomeIcon icon={faSun} color={"yellow"}/>} order={4}>{title}</Title>
+            </Action>
+        </div>
+    )
```

Comment:
Minor:  Missing semicolon.  I noticed there are some other pre-existing missing semi-colons up through the file too.